### ISSUE TITLE
[Feature Request] RCon support and chat logging with client ids

### ISSUE THREAD

**Issue by RaidMax** [R2Northstar/Northstar#76]
I'm the developer of an administration tool for Call of Duty https://github.com/RaidMax/IW4M-Admin and would like to add support for Northstar.

In order to add support, I would need the ability to remotely execute commands on dedicated servers and have chat messages logged with the client ids associated to them. (that's a bare minimum and the implementation details are flexible). I think having administration tool compatibility would be very beneficial to server owners and communities. 

It's possible the chat logging can be done with just mods, but I am not familiar enough with engine to verify.

**Comment by tittyfallfan69** [comment]
Please keep your token grabber out thx

**Comment by DxsSucuk** [comment]
Agreed.

**Comment by iCrazyBlaze** [comment]
There is already an RCon mod for Northstar

**Comment by RaidMax** [comment]
> There is already an RCon mod for Northstar

It’s not really RCon. It proxies client commands through a script to the server. It can only be used ingame and does not have proper authentication.

**Comment by yolarrydabomb** [comment]
> Please keep your token grabber out thx
> Agreed.

Maybe you both need to learn more of what IW4MAdmin does. before instantly thinking "OMG THIS THING IS A TOKEN GRABBER". IW4MAdmin is a very useful administrator server tool to keep abusive and toxic players out with lots of plugins support like map vote via in-game chat and personal server stats of kills and death. Even if it does grab some type of client id which you guys seem to call it a "token" Doesn't mean server admins able to steal your account. 🙄

**Comment by tittyfallfan69** [comment]
Keep your fascist hacks out of titanfall, losers.

**Comment by yolarrydabomb** [comment]
> Keep your fascist hacks out of titanfall, losers.
This tool helps protect cheaters not against cheaters. You moron.

**Comment by luigistyle** [comment]
> Keep your fascist hacks out of titanfall, losers.

Account made on the same day the issue was, nice troll account

**Comment by tittyfallfan69** [comment]
Be honest how many of you people supporting this are reddit mods?

**Comment by yolarrydabomb** [comment]
> Be honest how many of you people supporting this are reddit mods?

None. Good day.

**Comment by tittyfallfan69** [comment]
Sure seems like it if you're just waiting to respond so quickly. Did I touch a nerve or something? You're an authoritarian loser.

**Comment by HexChaos** [comment]
> Sure seems like it if you're just waiting to respond so quickly. Did I touch a nerve or something? You're an authoritarian loser.

Oooooooh someone's learn a new word but are unable to use it. Nice try buddy. 2/10 for trying

**Comment by yolarrydabomb** [comment]
> Sure seems like it if you're just waiting to respond so quickly. Did I touch a nerve or something? You're an authoritarian loser.

Nah just low IQ or troll people like you should never be on github that have 0 clue what they talking about. I rather not continue this drama any longer. I'm here to support RaidMax and his project and believe this be a good addition for Northstar that is all.

Feeding time for trolls is over. Get a life my man. lol

**Comment by tittyfallfan69** [comment]
Angry token grabber shills coming to brigade because the titanfall community doesn't want your censorship tools. I bet neither of you play the game. Go back to CoD, gaming nazis, the skies are ours.

**Comment by MerijnHendriks** [comment]
Please stop acting like children, this is a development zone not reddit or discord.

Having that said, I agree with @RaidMax that having at least minimal rcon support would be beneficial to server owners and its players.

**Comment by x3Karma** [comment]
There's no chat logging yet, remote execution might be possible though

**Comment by laundmo** [comment]
@RaidMax you might be interested in my fork of the software that I made mostly for personal use, it allows sending commands and receiving logs through raw TCP sockets. the logs include chat messages, and you can send console commands or whole squirrel scripts.

https://github.com/laundmo/NorthstarLauncher

**Comment by Vexbane** [comment]
I am grateful to all of you that are dedicating your time to working on this project for free, but understand this is a 'passion' project.  Which tends to lead people work on specific things they are passionate about.  I don't want to take away any free lance developer from working on specific parts of Northstar, but if there are any out there that might have the same passion as me for trying to get IW4MAdmin Rcon Tool Raidmax has developed to work with Northstar - I am willing to compensate you for your time to make changes necessary or at the very least try to discover how difficult it would be to implement / test - even if it can't come to fruition.  It might be a little less fun to work on than other things and hope a financial arrangement might help to make it worth someone's time.  

Respectfully - Discord: Vexbane#0701 

[11:56 AM] Vexbane: where you stand with iw4m + titanfall.  Im close to offering someone from northstar $ to work on integration with iw4m, if it comes to that - but need to know your commitment / interest in / willingness to work with someone if i hire them
[11:57 AM] RaidMax: All they need to do is implement an rcon system and chat logging
[11:57 AM] RaidMax: It’s pretty simple to integrate IW4MAdmin with that after
[11:57 AM] Vexbane: is that something that needs to happen on the back end with bob or any of the coder / devs helping to work on it?
[11:58 AM] Vexbane: who is "they" you refer to exactly
[11:59 AM] RaidMax: Any developer that has the ability to implement it can do it
[12:01 PM] RaidMax: It’s basically just opening a socket that listens for commands and rcon password that matches and the relays them to the game engine command processor and returns any output

**Comment by laundmo** [comment]
>  I am willing to compensate you for your time to make changes necessary

![image](https://user-images.githubusercontent.com/24855949/150691372-b077cf61-5ab5-4fbe-b8e3-8cec69bf0c5d.png)

**Comment by Vexbane** [comment]
> > I am willing to compensate you for your time to make changes necessary
> 
> ![image](https://user-images.githubusercontent.com/24855949/150691372-b077cf61-5ab5-4fbe-b8e3-8cec69bf0c5d.png)

donations and paid labor are different things.  I already read and was aware of your screenshot, but thank you.

**Comment by tittyfallfan69** [comment]
Vexbane you're such a loser lmao

**Comment by laundmo** [comment]
> Vexbane you're such a loser lmao

curb yo disrespect

**Comment by tittyfallfan69** [comment]
You can delete comments, but I'm not wrong.

**Comment by tittyfallfan6969** [comment]
You chuds can ban me, but I'm right. Everyone supporting this feature request is an authoritarian bootlicker.

**Comment by gaytranny** [comment]
wow this is a mess. you're kind of a faggot for wanting this in the mod btw definitely not going to just push more and more people away or anything like that. Judging by the comments still allowed the privileged to exist, I assume all the removed comments are just criticism?

**Comment by laundmo** [comment]
i mean, RCON is gonna happen. Thats inevitable, because server owners want to be able to manage their servers.

if you disagree with the RaidMax tool or whatever, maybe take that elsewhere, this thread is mostly asking about RCON

**Comment by tittyfallfan6969** [comment]
While I don't agree with the f slur that gaytanny used, they are correct, you are privileged and need to check it, unironically.

**Comment by tittyfallfan6969** [comment]
> i mean, RCON is gonna happen. Thats inevitable, because server owners want to be able to manage their servers.
> 
> if you disagree with the RaidMax tool or whatever, maybe take that elsewhere, this thread is mostly asking about RCON

I disagree with it being added to a game I care about, so ofc I would object to it here.

**Comment by laundmo** [comment]
@RaidMax 

Soo, update: theres a RCON PR now, https://github.com/R2Northstar/NorthstarLauncher/pull/100
For chat logging with client IDs, theres a feature now called "chathooks" which allows mods to receive chat messages. You could therefore easily make a mod that prints the chat messages in whatever format you would like to the log.
